### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by blessing-michael**
In this project I;
- Added Redux Toolkit
- Created a directory that will contain all your Redux logic
- Defined a slice of state for books
- Defined a slice of state for categories
- Ensured there are no linters error

### CODE REVIEW

File: src/redux/features/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = { categories: [] };
+
+const categoriesSlice = createSlice({
+  name: 'category',
+  initialState,
+  reducers: {
+    status: () => 'Under Construction',
+  },
```
Comment (AbelG101) [inline src/redux/features/categories/categoriesSlice.js]: - [ ] Ouch! this is a close one but as specified in the [project requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions_v2.md) please make sure to first check the status (from the action payload) before returning the string, take a look at the code snippet shown below:
```jsx
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
```

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @blessing-michael 👋, 

You're doing a good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Redux is configured properly ✔
- Proper use of createSlice ✔
- Correct use of Git Flow ✔
- All linter checks are passing ✔
- Descriptive PR title with a great summary ✔

## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/features/categories/categoriesSlice.js (hunk 1 above)
Comment (blessing-michael) [reply src/redux/features/categories/categoriesSlice.js]: @AbelG101
 Hello Abel,
I'm Blessing. Thank you for the review. Please kindly throw more light as I'm confused on where to add the suggested code you requested. I look forward to hearing from you. Thank you.

**Review by Alusp — APPROVED** [review]
Hi @blessing-michael, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
congrats

✅ Implemented tests are passing
✅ Write unit tests using Rspec
✅ Add unit tests for all the methods in all the classes
✅ Understand the concept of state management.
✅ Use store, actions, and reducers in React.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Alusp in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/features/categories/categoriesSlice.js (hunk 1 above)
Comment (AbelG101) [reply src/redux/features/categories/categoriesSlice.js]: - I apologize If I wasn't clear in my comments previously, but you just need to update the status reducer in the category slice so you'd have something like the below
```jsx
status: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
},
```

**Pull request merged**